store/blockstore.go

```diff
@@ -0,0 +1,124 @@
+package store
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/gob"
```

Comment:
Bruh, why `gob`?

Reply:
It works out of the box, without external deps. I know that it will be replaced in the future anyways, so it's the easiest solution.

Reply:
I think that is OK, can you open a tracking issue for replacing this with another encoding format (yet to be decided).

Reply:
tracking issue: https://github.com/lazyledger/optimint/issues/45